### PR TITLE
Tpetra: batchedApply() prototype

### PR DESCRIPTION
**Pull request by csiefer2**
Tests: RHEL6

### CODE REVIEW

File: packages/tpetra/core/src/Tpetra_Apply_Helpers.hpp [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+  /// \param X [in]        - Tpetra::MultiVector or Tpetra::Vector object.
+  /// \param Y [out]       - [std::vector|Teuchos::Array|Teuchos::ArrayRCP] of Tpetra::MultiVector or Tpetra::Vector objects.
+  ///                        These must have the same number of vectors as X.
+  template <class MatrixArray, class MultiVectorArray> 
+  void batchedApply(const MatrixArray &Matrices, 
+                    const typename std::remove_pointer<typename MultiVectorArray::value_type>::type &X,
+                    MultiVectorArray &Y,
+                    typename std::remove_pointer<typename MatrixArray::value_type>::type::scalar_type alpha = Teuchos::ScalarTraits< typename std::remove_pointer<typename MatrixArray::value_type>::type::scalar_type>::one(),
+                    typename std::remove_pointer<typename MatrixArray::value_type>::type::scalar_type beta  = Teuchos::ScalarTraits< typename std::remove_pointer<typename MatrixArray::value_type>::type::scalar_type>::zero() ) {
+    Tpetra::Details::ProfilingRegion regionImport ("Tpetra::batchedApply");
```
Comment (cgcgcg) [inline packages/tpetra/core/src/Tpetra_Apply_Helpers.hpp:72]: `regionImport` needs renaming to avoid clash with the other timer below.

Comment (csiefer2) [reply packages/tpetra/core/src/Tpetra_Apply_Helpers.hpp:72]: Good catch!

**Comment by csiefer2** [comment]
@trilinos/framework There's a randomly failing STK test on this one.  Not me, I swear

**Comment by alanw0** [comment]
@csiefer2 Yep, that one's not you. I'll push up a change as soon as I can, to robustify that test.

**Pull request merged**